src/js/components/ToggleButtonGroup/ToggleButtonGroup.js

```diff
@@ -0,0 +1,182 @@
+import { Box } from '../Box';
+import { Keyboard } from '../Keyboard';
+import { ToggleButtonGroupPropTypes } from './propTypes';
+import { StyledButton } from './StyledToggleButtonGroup';
+
+const useControlled = ({ prop, defaultProp, onChange = () => {} }) => {
+  const [uncontrolledProp, setUncontrolledProp] = useState(defaultProp);
+  const controlled = prop !== undefined;
+  const value = controlled ? prop : uncontrolledProp;
+  const handleChange = useCallback(onChange, [onChange]);
+
+  const setValue = useCallback(
+    (nextValue) => {
+      // only update internal value in uncontrolled cases
+      if (!controlled) {
+        setUncontrolledProp(nextValue);
+      }
+      handleChange(nextValue);
+    },
+    [controlled, setUncontrolledProp, handleChange],
+  );
+
+  return [value, setValue];
+};
+
+const ToggleButtonGroup = ({
+  defaultValue,
+  multiple,
+  options,
+  onChange,
+  value: valueProp,
+  ...rest
+}) => {
+  const [value = multiple ? [] : '', setValue] = useControlled({
+    prop: valueProp,
+    defaultProp: defaultValue,
+    onChange,
+  });
+  const theme = useContext(ThemeContext);
+  const ref = useRef();
+  const buttonRefs = useRef([]);
+
+  const values = options.map((option) =>
+    typeof option === 'object' ? option.value : option,
+  );
+
+  const getFocusableIndex = useCallback(() => {
+    let defaultIndex = 0;
+    if (value.length) {
+      // set earliest button that's part of value to active
+      // assume that value might not be ordered the same as options
+      defaultIndex = values.indexOf(
+        values.find((option) => value.includes(option)),
+      );
+    }
+    return defaultIndex;
+  }, [value, values]);
+
+  const [focusableIndex, setFocusableIndex] = useState(() =>
+    getFocusableIndex(),
+  );
+
+  const onNext = (e) => {
+    // prevent page scroll
+    e.preventDefault();
+    const nextIndex =
+      focusableIndex + 1 <= options.length - 1 ? focusableIndex + 1 : 0;
+    setFocusableIndex(nextIndex);
+    buttonRefs.current[nextIndex].focus();
+  };
+
+  const onPrevious = (e) => {
+    // prevent page scroll
+    e.preventDefault();
+    const nextIndex =
+      focusableIndex - 1 >= 0 ? focusableIndex - 1 : options.length - 1;
+    setFocusableIndex(nextIndex);
+    buttonRefs.current[nextIndex].focus();
+  };
+
+  const handleToggle = (option, active) => {
+    let nextValue;
+    if (multiple) {
+      if (Array.isArray(value)) {
+        nextValue = active
+          ? value.filter((item) => item !== option)
+          : [...value, option];
+      } else {
+        nextValue = active ? [] : [option];
+      }
+    } else {
+      nextValue = active ? '' : option;
+    }
+    setValue(nextValue);
+  };
+
+  return (
+    <Keyboard
+      onUp={onPrevious}
+      onDown={onNext}
+      onLeft={onPrevious}
+      onRight={onNext}
+    >
+      <Box
+        ref={ref}
+        direction="row"
+        alignSelf="start"
+        role="group"
```

Comment:
I think this role will need to be conditional based on whether it is in a radio vs. multiple mode.

Per spec `role="radio"` must be contained in a `role="radiogroup"` ([reference](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/radio_role#description))

When in `multiple`, `role="group"` is appropriate.

Reply:
Done.